packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { ulStyleOverrides, LEFT_RIGHT_OFFSET } from './styles';
+
+const sideNavLabelStyle = css`
+  display: block;
+  color: ${uiColors.green.base};
+  font-size: 12px;
+  font-weight: bold;
+  text-transform: uppercase;
+  padding: 5px ${LEFT_RIGHT_OFFSET}px 5px ${LEFT_RIGHT_OFFSET}px;
+`;
+
+interface SideNavGroupProps {
+  /**
+   * Content that will appear inside of SideNavGroup component.
+   */
+  children: React.ReactNode;
+  /**
+   * ClassName that will be applied to root SideNavGroup element.
+   */
+  className?: string;
+  /**
+   * Header text displayed above the SideNavGroup contents.
+   */
+  headerText?: string;
+  /**
+   * ClassName that will be applied to SideNavGroup's header element.
+   */
+  headerClassName?: string;
+}
+
+/**
+ * # SideNavGroup
+ *
+ * ```
+<SideNavGroup headerText="Section Header">
+  <SideNavItem>
+    Link Content
+  </SideNavItem>
+</SideNavGroup>
+ * ```
+ * @param props.children Content that will appear inside of SideNavGroup component.
+ * @param props.className ClassName that will be applied to root SideNavGroup element.
+ * @param props.headerText Header text displayed above the SideNavGroup contents.
+ * @param props.headerClassName ClassName that will be applied to SideNavGroup's header element.
+ *
+ */
+function SideNavGroup({
+  children,
+  className,
+  headerText,
+  headerClassName,
+  ...rest
+}: SideNavGroupProps) {
+  return (
+    <li className={className} {...rest}>
+      {headerText && (
+        <div className={cx(sideNavLabelStyle, headerClassName)} role="heading">
```

Comment:
We should probably use a more semantic tag than `div` here. Perhaps something like `label` or `h4`?

Reply:
Additionally, I think we get `role="header"` for free that way.

Reply:
:+1 @DesignerDave

Reply:
I wanted to use a h tag, but didn't know what level you all use for what. H4 it is!

Reply:
I'm sure we'll have stricter guidelines in the future, but for the time being this should do!